### ISSUE TITLE
[Paper Note] On Surprising Effectiveness of Masking Updates in Adaptive Optimizers, Taejong Joo+, arXiv'26, 2026.02

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#4621]
# URL
- https://arxiv.org/abs/2602.15322
# Authors
  - Taejong Joo
  - Wenhan Xia
  - Cheolmin Kim
  - Ming Zhang
  - Eugene Ie
# Abstract
  - Training large language models (LLMs) relies almost exclusively on dense adaptive optimizers with increasingly sophisticated preconditioners. We challenge this by showing that randomly masking parameter updates can be highly effective, with a masked variant of RMSProp consistently outperforming recent state-of-the-art optimizers. Our analysis reveals that the random masking induces a curvature-dependent geometric regularization that smooths the optimization trajectory. Motivated by this finding, we introduce Momentum-aligned gradient masking (Magma), which modulates the masked updates using momentum-gradient alignment. Extensive LLM pre-training experiments show that Magma is a simple drop-in replacement for adaptive optimizers with consistent gains and negligible computational overhead. Notably, for the 1B model size, Magma reduces perplexity by over 19\% and 9\% compared to Adam and Muon, respectively.
# Translation (by gpt-5-nano)
- 大規模言語モデル（LLMs）のトレーニングは、ほぼ完全に密な適応型オプティマイザに依存しており、前処理行列がますます高度化している。これに挑戦するべく、パラメータ更新をランダムにマスキングすることが非常に効果的であることを示す。マスクされたRMSPropの派生形は、最近の最先端オプティマイザを一貫して上回る。我々の分析は、ランダムマスキングが曲率依存の幾何学的正則化を誘発し、最適化の軌道を滑らかにすることを明らかにする。 この発見に動機づけられ、Momentum-aligned gradient masking（Magma）を導入する。これは、モーメントと勾配の整合性を用いてマスクされた更新を調整する。 広範なLLMの事前学習実験は、Magmaが適応型オプティマイザの単純なドロップイン置換であり、一貫した改善とほとんど計算オーバーヘッドがないことを示している。 特に1Bモデルサイズでは、MagmaはAdamおよびMuonと比較して、それぞれ19%超および9%のパープレキシティを低減する。
# Summary (by gpt-4o-mini)
- 大規模言語モデルのトレーニングでは、密な適応型オプティマイザが主流であるが、ランダムマスキングを用いたパラメータ更新が効果的であることを示す。マスクされたRMSPropが最新オプティマイザを上回ることを明らかにし、Momentum-aligned gradient masking（Magma）を導入。Magmaはモーメントと勾配の整合性を利用したマスク更新を行い、一貫した性能向上を達成し、計算オーバーヘッドがほとんどない。特に1Bモデルサイズで、MagmaはAdamおよびMuonに対してそれぞれ19%超および9%のパープレキシティ改善を示す。

**Comment by AkihikoWatanabe** [comment]
元ポスト:https://x.com/rosinality/status/2023974107399024769?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
元ポスト:https://x.com/hillbig/status/2024252068958261624?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q
